### ISSUE TITLE
Creating a profile changes the language of Joplin

### ISSUE THREAD

**Issue by JackGruber** [laurent22/joplin#9960]
### Operating system

Windows

### Joplin version

2.14.14

### Desktop version info

Joplin 2.14.14 (prod, win32)

Client ID: 0e33609c84614b3092102ff00887b36e
Sync Version: 3
Profile Version: 45
Keychain Supported: Yes

Revision: ecb6ac3

### Current behaviour

1. Change your Joplin language other than the operating system language
2. Create a new profile
3. The language for the new profile is the same as for the operating system 
4. Change back to your original profile 
5. The language has now been changed to that of the operating system here too

### Expected behaviour

The language should not be changed automatically when creating a new Joplin profile

**Comment by abansal15** [comment]
hello @laurent22 could you please asssign this issue to me

**Comment by roman-r-m** [comment]
We do not assign issues

**Comment by HahaBill** [comment]
I have the same issue! (macOS Sonoma, 14.3.1). I will try to have a look once I am done with my PR :)

Other contributors and GSoC candidates, feel free to also have a look and solve the issue!

**Comment by PrathameshSonje** [comment]
I'm a little confused about what the expected behavior should be,

If I'm not mistaken, the expected behavior is that the language set in the default profile should apply to new profiles by default.

Default profile - Arabic
New profile - ENG (actual behavior)
New Profile - Arabic (expected)

**Comment by JackGruber** [comment]
If the language was changed in the default profile, it should not be changed by creating a new profile to the system default language.

**Comment by cagnusmarlsen** [comment]
So I looked into this issue and I think I understand what the problem is. When a new profile is created, this line
```typescript
const locale = shim.detectAndSetLocale(Setting);
``` 
in
 `/packages/lib/BaseApplication.ts` sets the `locale` to the system default language and because `locale` is a global property (`isGlobal: true`), it gets changed in the default profile as well as all other profiles. This is what changes the language of the default profile to the system default.

A fix for this would be to not set `locale` when a new profile is created, and instead create the new profile with the same `locale` value. This ensures that the language set in the default profile is applied globally to all profiles and a change of languages applies to every profile. I've tested the app with these changes and it seems to be working the way it should be.

I could create a PR with these changes if this seems like the right way to go.
